enhancements/host-network-configuration.md

```diff
@@ -0,0 +1,494 @@
+
+Consider a lab environment where DHCP services either aren’t available at all,
+or the person trying out an OpenShift cluster does not have access to it to
+configure DHCP address reservations.  As long as the cluster administrator has
+a pool of addresses they are allowed to assign manually, a cluster with fully
+static network configuration would allow them to install OpenShift without
+needing to work with the network administrator on DHCP server configuration.
+
+* Must be able to override default RHCOS behavior of requiring DHCP to be
+  present on at least one interface
+* Must be able to configure static network configuration for every interface
+  used by the cluster
+
+
+## Existing Configuration Interfaces
+
+### Summary
+
+First, here’s a summary of how an administrator may configure host networking
+in a cluster.  More details about each interface are in the following sections.
+
+#### First Boot
+
+When an RHCOS host first boots, the initramfs environment must be able to reach
+the [Machine Config Server
+(MCS)](https://github.com/openshift/machine-config-operator/blob/master/docs/MachineConfigServer.md)
+for Ignition to succeed.  The MCS starts by running on the Boostrap host while
+the first hosts are being deployed.  Ignition will attempt to reach the Machine
+Config Server via an included URL. By default, dracut will attempt to do DHCPv4
+and DHCPv6 on all interfaces and wait for one of them to work.  To change this
+behavior, an admin can:
+
+* Change the `ip=<...>` kernel argument(s) via PXE configuration or at the grub
+  command line during boot.  If doing a bare metal install, it is also possible
+  to edit the arguments baked into the image using a tool such as
+  `virt-customize`.  Note that this is really the interface of last resort and
+  shouldn’t be used unless absolutely required.
+* If at least one network has DHCP to allow dracut to succeed, custom ignition
+  additions can configure additional networking
+* If using the RHCOS live ISO/PXE ramdisk environment with the
+  coreos-installer, use the `nmtui` Network Manager text UI to configure
+  networking before proceeding.  Those settings can be automatically propogated
+  to the installed host by using the `--copy-network` option to the
+  `coreos-installer`.
+
+#### Further Network Customization
+
+Beyond the minimum needed to bootstrap the cluster, further host network
+configuration customizations can be applied in the following ways:
+
+* Customize the ignition configuration at install time to include more
+  NetworkManager configuration files
+* Create `MachineConfig` resources with NetworkManager configuration files and
+  have them applied by the installer at install time, or apply new
+  `MachineConfig` resources to the cluster via the API any time post-install
+
+### Dracut in initfamfs
+
+In the initramfs environment at boot time,
+[dracut](https://github.com/dracutdevs/dracut) is the first tool that sets up
```

Comment:
maybe clarify that this is an "upstream" tool, eg, used by RHEL and Fedora as well (as compared to ignition)

Reply:
done!